### PR TITLE
first pass at defining local and remote nodes

### PR DESCRIPTION
**Pull request by LiranCohen**

### CODE REVIEW

File: spec/spec.md [hunk 1]
```diff
@@ -69,6 +69,14 @@ distributed ledger).
 
 <img src="images/topology.svg" style="display: block; margin: 0 auto; padding: 1em 0; width: 100%; max-width: 1000px;" />
 
+### Relay Nodes
+
+Relay nodes act as publically accessable service endpoints for DWN infrastructure that may be advertized within a user's DID document.  Relay nodes do not contain private key material and as such do not encrypt, sign, or authorize access to data.  Relay nodes queue and store incoming messages which need to be accessed by a user's local node, and also serve as remote data stores for data assocaited with the DID controled by a user's local node. A relay node may answer a request without requiring interaction from a user's local node if those permissions already exist.
```
Comment (TallTed) [inline spec/spec.md:74]:
```suggestion
Relay nodes act as publicly accessible service endpoints for DWN infrastructure that may be advertised within a user's DID document.  Relay nodes do not contain private key material and as such do not encrypt, sign, or authorize access to data.  Relay nodes queue and store incoming messages which need to be accessed by a user's local node, and also serve as remote data stores for data associated with the DID controlled by a user's local node. A relay node may answer a request without requiring interaction from a user's local node if those permissions already exist.
```

**Comment by LiranCohen** [comment]
Thanks for the grammar corrections @TallTed!

**Comment by greyman888** [comment]
Sorry if this is not the right place for this feedback. Please point me in the right direction if so.

I'm trying to build up an understanding from the perspective of implementation architecture, responsibility & control. 

It's great that you clarify that the local node operates on the DID controller's device.

It would help my understanding a lot if you contrasted that with where is the relay node operating, who operates it and why would they do this?

Given that the relay node is running on a server and always available to answer requests, the implementation question I then have is: who sets up and controls the relay node?

- Is it the DID controller responsibility (Alice, Bob)?
- Is it an unrelated third party?

I feel like I'm fundamentally misunderstanding the design. 

If the end user is responsible for the relay node, does that mean they need to pay to spin up their own relay node on something like AWS?

If the relay node is controlled by a third party who is not Alice or Bob then it would be great if they appeared in the description with the service they are providing. I would then start thinking about how this service would be run.

**Comment by serverwentdown** [comment]
@greyman888 my interpretation of the spec is that the "relay" node must be provided by the user through the DID. I would imagine that there will be an ecosystem of providers (like Solid does) that provide users with "relay" nodes.

**Comment by OR13** [comment]
@csuwildcat who is taking point on code reviews here?

**Comment by andorsk** [comment]
@LiranCohen let's discuss this during the DIF call today and see what the right course of action is to move this forward.

**Pull request closed without merging**